src/k8s/pkg/component/component.go

```diff
@@ -142,28 +170,32 @@ func (h *helmClient) isComponentEnabled(name, namespace string) (bool, error) {
 
 // List lists the status of each k8s component.
 func (h *helmClient) List() ([]Component, error) {
-	actionConfig, err := h.initializeHelmClientConfig()
+	actionConfig, err := h.initializer.InitializeHelmClientConfig()
 	if err != nil {
 		return nil, fmt.Errorf("failed to initialize Helm client configuration: %w", err)
 	}
 
 	list := action.NewList(actionConfig)
 	releases, err := list.Run()
+
 	if err != nil {
 		return nil, fmt.Errorf("failed to list components: %w", err)
 	}
 
-	allComponents := make([]Component, len(h.config))
+	allComponents := make([]Component, 0)
```

Comment:
A bit of Go nerding:
you can also specify the capacity of the backing array and still set a size of 0 for the slice:
```
allComponents := make([]Component, 0, len(h.config)
```
See: https://goplay.tools/snippet/-pCq4-6OwX8

This is totally crucial for the performance of the k8s-snap and not premature optimization! ;)

Reply:
Ah, good to know! Learnt something new again :)